packages/troika-three-text/src/Typesetter.js

```diff
@@ -677,6 +689,25 @@ export function createTypesetter(resolveFonts, bidi) {
     }
   }
 
+  function fillSurrogateCaretPositions(caretPositions, charStartIndex, charCount) {
```

Comment:
This could be simplified to just a single `caretPositions.copyWithin()` call. (Edit: if this function stays; see other comment.)